openmls/src/group/mls_group/application.rs

```diff
@@ -15,31 +15,43 @@ impl MlsGroup {
         &mut self,
         backend: &impl OpenMlsCryptoProvider,
         message: &[u8],
-    ) -> Result<MlsMessageOut, MlsGroupError> {
+    ) -> Result<MlsMessageOut, CreateMessageError> {
         if !self.is_active() {
-            return Err(MlsGroupError::GroupStateError(
+            return Err(CreateMessageError::GroupStateError(
                 MlsGroupStateError::UseAfterEviction,
             ));
         }
         if !self.proposal_store.is_empty() {
-            return Err(MlsGroupError::GroupStateError(
+            return Err(CreateMessageError::GroupStateError(
                 MlsGroupStateError::PendingProposal,
             ));
         }
 
-        let credential = self.credential()?;
+        let credential = self
+            .credential()
+            // We checked we are in the right group state before
+            .map_err(|_| LibraryError::custom("Wrong group state"))?;
```

Comment:
How does failing to fetch a credential indicate a wrong group state? That's not quite clear to me.

Reply:
The own credential is removed after eviction, that's why you don't have access to it anymore:

```rust
    pub fn credential(&self) -> Result<&Credential, MlsGroupStateError> {
        if !self.is_active() {
            return Err(MlsGroupStateError::UseAfterEviction);
        }
        let tree = self.group.treesync();
        Ok(tree
            .own_leaf_node()
            .map_err(|_| LibraryError::custom("Own leaf node missing"))?
            .key_package()
            .credential())
    }
 ```